include/librealsense2/dds/dds-device.cpp

```diff
@@ -36,7 +74,180 @@ class dds_device::impl
         , _participant( participant )
     {
         (void)guid;
+
+        create_notifications_reader();
+        if( !init() )
+        {
+            throw std::runtime_error("failed getting sensors data from device: " + info.topic_root);
+        }
+        LOG_DEBUG("Device" << _info.topic_root << " was initialize successfully" );
+    }
+
+private:
+
+    void create_notifications_reader()
+    {
+        // CREATE THE SUBSCRIBER
+        _subscriber = DDS_API_CALL( _participant->get()->create_subscriber( SUBSCRIBER_QOS_DEFAULT, nullptr ) );
+        if( _subscriber == nullptr )
+        {
+            throw std::runtime_error( "Error creating a DDS subscriber" );
+        }
+
+        // CREATE TOPIC TYPE
+        eprosima::fastdds::dds::TypeSupport topic_type(
+            new librealsense::dds::topics::device::notification::type );
+
+        // REGISTER THE TYPE
+        DDS_API_CALL( topic_type.register_type( _participant->get() ) );
+
+
+
+        std::string topic_name
+            = librealsense::dds::topics::device::notification::construct_topic_name( _info.topic_root );
+
+        // CREATE THE TOPIC
+        _topic = DDS_API_CALL( _participant->get()->create_topic( topic_name,
+                                                                  topic_type->getName(),
+                                                                  TOPIC_QOS_DEFAULT ) );
+
+        // CREATE THE READER
+        DataReaderQos rqos = DATAREADER_QOS_DEFAULT;
+        rqos.data_sharing().off();
+        rqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
+        rqos.durability().kind = VOLATILE_DURABILITY_QOS;
+        rqos.history().kind = KEEP_LAST_HISTORY_QOS;
+        rqos.history().depth = 10;
+        rqos.endpoint().history_memory_policy
+            = eprosima::fastrtps::rtps::PREALLOCATED_WITH_REALLOC_MEMORY_MODE;
+        _reader = DDS_API_CALL( _subscriber->create_datareader( _topic, rqos, nullptr ) );
+        if( _reader == nullptr )
+        {
+            throw std::runtime_error( "Error creating a DDS reader for " + topic_name );
+        }
+
+        LOG_DEBUG( "topic '" << topic_name << "' created" );
+    }
+
+    bool init()
```

Comment:
```c++
    // The device exists - we know about it - but is unusable until we get details (sensors, profiles, etc.) and
    // initialization is complete. This initialization depends on several messages from the server, and may take some
    // time.
    //
    // Returns true if initialization was successful.
```
I think maybe we want to emphasize the waiting. Maybe call it `wait_for_initialization()`?

Also, this is called from the ctor -- so the waiting is hidden and happen right in the thread that created the device. E.g., the dds_device_watcher's main loop will actually wait for each device to be initialized before continuing. Is this what we want?

Reply:
Done - LRS-473